libsolidity/analysis/OverrideChecker.cpp

```diff
@@ -584,35 +584,35 @@ void OverrideChecker::checkOverride(OverrideProxy const& _overriding, OverridePr
 				"Overridden " + _overriding.astNodeName() + " is here:"
 			);
 
-		// This is only relevant for a function overriding a function.
-		if (_overriding.isFunction())
-		{
-			// Stricter mutability is always okay except when super is Payable
-			if ((
+		// Stricter mutability is always okay except when super is Payable
+		if (
+			(_overriding.isFunction() || _overriding.isVariable()) &&
+			(
 				_overriding.stateMutability() > _super.stateMutability() ||
 				_super.stateMutability() == StateMutability::Payable
-				) &&
-				_overriding.stateMutability() != _super.stateMutability()
-			)
-				overrideError(
-					_overriding,
-					_super,
-					6959_error,
-					"Overriding function changes state mutability from \"" +
-					stateMutabilityToString(_super.stateMutability()) +
-					"\" to \"" +
-					stateMutabilityToString(_overriding.stateMutability()) +
-					"\"."
-				);
+			) &&
+			_overriding.stateMutability() != _super.stateMutability()
+		)
+			overrideError(
+				_overriding,
+				_super,
+				6959_error,
+				"Overriding " +
+				_overriding.astNodeName() +
+				" changes state mutability from \"" +
+				stateMutabilityToString(_super.stateMutability()) +
+				"\" to \"" +
+				stateMutabilityToString(_overriding.stateMutability()) +
+				"\"."
+			);
 
-			if (_overriding.unimplemented() && !_super.unimplemented())
-				overrideError(
-					_overriding,
-					_super,
-					4593_error,
-					"Overriding an implemented function with an unimplemented function is not allowed."
-				);
-		}
+		if (_overriding.isFunction() && _overriding.unimplemented() && !_super.unimplemented())
```

Comment:
This might not be needed as the proxy always returns `false` for state variables